tactics/hints.ml

```diff
@@ -1384,7 +1441,30 @@ let prepare_hint check env init (sigma,c) =
     let diff = Univ.ContextSet.diff (Evd.universe_context_set sigma) (Evd.universe_context_set init) in
     (c', diff)
 
+let warn_non_local_section_hint =
+  CWarnings.create ~name:"non-local-section-hint" ~category:"automation"
+    (fun () -> strbrk "This hint is not local but will disappear when the section is closed.")
```

Comment:
What about expanding the message to explain "because it depends on a section variable"?

Reply:
Yes, for now this is a POC that I wrote in a hurry, but the message should be more explicit indeed.